pkg/reconciler/pipelinerun/pipelinerun_test.go

```diff
@@ -1307,15 +1307,15 @@ status:
 	prt := newPipelineRunTest(t, d)
 	defer prt.Cancel()
 
-	reconciledRun, clients := prt.reconcileRun("foo", "test-pipeline-missing-results", []string{}, true)
+	reconciledRun, clients := prt.reconcileRun("foo", "test-pipeline-missing-results", []string{}, false)
 	if reconciledRun.Status.CompletionTime == nil {
 		t.Errorf("Expected a CompletionTime on invalid PipelineRun but was nil")
 	}
 
-	// The PipelineRun should be marked as failed due to InvalidTaskResultReference.
+	// The PipelineRun should be marked as failed
```

Comment:
I was also concerned about not showing the exact reason for the failure. When I checked in the implementation inside the `GetPipelineConditionStatus` function, it gave status based on `failed`, and `success`. So it is more generalized. 
The status will not be available at the TaskRun level as TaskRun will not be created.
Do you have any suggestions to deal with this in a better way so that we can have a proper reason for the failure? 
Earlier we were failing right away so we were getting the exact error.

Reply:
One idea that I can think of is to have `ValidationFailed` as a map where it will store failed tasks along with the reason and inside the `GetPipelineConditionStatus` function we can have a check for the ValidationFailed and add the status accordingly.
If this looks good, I can try the implementation.

Reply:
One question that comes with this approach is when setting the reason in `GetPipelineConditionStatus` is how to decide the reason for the multiple Validation Failed Task. 
May be we can set this to a generic error like https://github.com/tektoncd/pipeline/blob/main/pkg/apis/pipeline/v1/pipelinerun_types.go#L385